### ISSUE TITLE
Bot is not pulling DNS Information

### ISSUE THREAD

**Issue by MattTatoczenko** [MattTatoczenko/AppServiceDomainChecker#3]
Getting the following a lot recently:

Thank you for confirming all of the information in regards to your App Service and custom hostname.

Let's pull some information on the hostname entered.

I was unable to pull DNS information on the A records for your hostname.

Let me try to get the DNS information I wasn't able to get before.

Let's pull some information on the hostname entered.

I was unable to pull DNS information on the A records for your hostname.

Let me try to get the DNS information I wasn't able to get before.

Let's pull some information on the hostname entered.

I am sorry. I am having a lot of issues with pulling DNS information. Please try to check the hostname information later. When you are ready to check the domain again, press anything to restart.

Same code seems to work locally. Will keep digging. For now, seem to be getting this exception in the code:
w3wp.exe Error: 0 : Error on dns query: System.Net.Sockets.SocketException (0x80004005): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
   at System.Net.Sockets.Socket.ReceiveFrom(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags, EndPoint& remoteEP)
   at ARSoft.Tools.Net.Dns.DnsClientBase.QueryByUdp(DnsClientEndpointInfo endpointInfo, Byte[] messageData, Int32 messageLength, IPAddress& responderAddress)

Will need to figure out how to configure System.Net tracing in a Bot Service/Function and get that running to dig into the issues with whatever endpoint it queries.

**Comment by MattTatoczenko** [comment]
Potentially fixed with https://github.com/MattTatoczenko/AppServiceDomainChecker/commit/bd9ef524adfda9f843a2ef2ad04df1358d43092b

**Comment by MattTatoczenko** [comment]
Issue was with this code:
`IDnsResolver resolver = new DnsStubResolver();`
`List<ARecord> aRecords = DnsResolverExtensions.Resolve<ARecord>(resolver, appService.CustomHostname, RecordType.A, RecordClass.Any);`

For some reason, the DnsStubResolver was not properly connecting to the DNS servers used by default in the Azure Bot Service.

Instead, I changed to using the following:
`DnsMessage dnsMessage = DnsClient.Default.Resolve(DomainName.Parse(appService.CustomHostname), RecordType.A);`
A direct DNS client was able to resolve the hostname information. Updated all of the checkers to use this DNS Client method, which is working in Azure and in my testing.

All code was pushed in https://github.com/MattTatoczenko/AppServiceDomainChecker/commit/bd9ef524adfda9f843a2ef2ad04df1358d43092b that fixed this.
